hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/NodeStatusUpdaterImpl.java

```diff
@@ -783,15 +783,11 @@ public void removeVeryOldStoppedContainersFromCache() {
           break;
         }
         if (!context.getContainers().containsKey(cid)) {
-          ApplicationId appId =
-              cid.getApplicationAttemptId().getApplicationId();
-          if (isApplicationStopped(appId)) {
```

Comment:
The current code will also delete all delegation tokens in the application, including those used for log aggregation, causing log aggregation to fail.

Reply:
Can you elaborate? Are you saying there is a problem with these changes? Or something else wrong in the existing code?

Reply:
I carefully reviewed the code again and there should be no problem of aggregation failure.

By the way.

Aggregate containers that take more than 30 minutes (or longer) to complete onto HDFS in advance, rather than waiting for the job to complete.

Is this implementation better than the current one?